src/services/CleanupService.ts

```diff
@@ -222,79 +205,154 @@ export const cleanCookies = async (
   });
 };
 
-/** This will use the browsingData's hostname attribute to delete any extra browsing data */
+// Cleanup of all cookies for domain.
+export const clearCookiesForThisDomain = async (
+  state: State,
+  tab: browser.tabs.Tab,
+) => {
+  const hostname = getHostname(tab.url);
+  const firstPartyIsolate = await isFirstPartyIsolate();
+  const getCookies = (await browser.cookies.getAll(
+    returnOptionalCookieAPIAttributes(state, {
+      domain: hostname,
+      storeId: tab.cookieStoreId,
+    }, firstPartyIsolate),
+  ));
+  const cookies = getCookies.filter(c => c.name !== LSCLEANUPNAME);
+
+  if (cookies.length > 0) {
+    let cookieDeletedCount = 0;
+    for (const cookie of cookies) {
+      const r = await browser.cookies.remove(returnOptionalCookieAPIAttributes(state, {
+        firstPartyDomain: cookie.firstPartyDomain,
+        name: cookie.name,
+        storeId: cookie.storeId,
+        url: prepareCookieDomain(cookie),
+      }, firstPartyIsolate) as {
+        // Fix type error with undefineds with cookies.remove
+        url: string;
+        name: string;
+      });
+      if (r !== null) cookieDeletedCount += 1;
```

Comment:
`if (!r)` ?

Reply:
yeah I can change that.  I was just following MDN where it says it will return null if it wasn't removed.

Reply:
so I guess it should be `if (r)` instead.

Reply:
resolved via 6a1a285